File: app/src/main/java/com/xpayworld/payment/PaymentApplication.java
package com.xpayworld.payment;
import android.app.Application;
import com.xpayworld.payment.di.component.ApplicationComponent;
import com.xpayworld.payment.di.component.DaggerApplicationComponent;
import com.xpayworld.payment.di.module.ApplicationModule;


public class PaymentApplication extends Application {

    private ApplicationComponent mAppComponent;

    @Override
    public void onCreate() {
        super.onCreate();
        mAppComponent = DaggerApplicationComponent.builder()
                .applicationModule(new ApplicationModule(this))
                .build();
        mAppComponent.inject(this);

    }
    public ApplicationComponent getAppComponent() {
        return mAppComponent;
    }
}

File: app/src/main/java/com/xpayworld/payment/PaymentApplication.kt
package com.xpayworld.payment

import android.app.AlertDialog
import android.app.Application
import android.content.DialogInterface
import com.xpayworld.payment.di.component.ApplicationComponent
import com.xpayworld.payment.di.component.DaggerApplicationComponent
import com.xpayworld.payment.di.module.ApplicationModule
import com.xpayworld.payment.util.RootUtil
import com.xpayworld.payment.util.isProbablyAnEmulator


class PaymentApplication : Application() {

    var appComponent: ApplicationComponent? = null
        private set

    override fun onCreate() {
        super.onCreate()
        appComponent = DaggerApplicationComponent.builder()
                .applicationModule(ApplicationModule(this))
                .build()
        appComponent!!.inject(this)

    }
}
